src/truncate.jl

```diff
@@ -61,6 +60,29 @@ logccdf(d::Truncated, x::Float64) = x <= d.lower ? 0.0 :
 quantile(d::Truncated, p::Float64) = quantile(d.untruncated, d.lcdf + p * d.tp)
 
 
+pdf(d::Truncated, x::Int64) = d.lower <= x <= d.upper ? pdf(d.untruncated, x) / d.tp : 0.0
+
+logpdf(d::Truncated, x::Int64) = d.lower <= x <= d.upper ? logpdf(d.untruncated, x) - d.logtp : -Inf
+
+cdf(d::Truncated, x::Int64) = x <= d.lower ? 0.0 : 
+                             x >= d.upper ? 1.0 :
+                             (cdf(d.untruncated, x) - d.lcdf) / d.tp
+
+logcdf(d::Truncated, x::Int64) = x <= d.lower ? -Inf :
+                                x >= d.upper ? 0.0 :
+                                log(cdf(d.untruncated, x) - d.lcdf) - d.logtp
+
+ccdf(d::Truncated, x::Int64) = x <= d.lower ? 1.0 : 
+                              x >= d.upper ? 0.0 :
+                              (d.ucdf - cdf(d.untruncated, x)) / d.tp
+
+logccdf(d::Truncated, x::Int64) = x <= d.lower ? 0.0 :
+                                 x >= d.upper ? -Inf :
+                                 log(d.ucdf - cdf(d.untruncated, x)) - d.logtp 
+
+
+quantile(d::Truncated, p::Int64) = quantile(d.untruncated, d.lcdf + p * d.tp)
+
```

Comment:
When I try to change these to `Real` I get a complaint about ambiguity with definitions in `univariates.jl`


```julia
julia> using Distributions
Warning: New definition 
    pdf(Truncated{S<:ValueSupport,D<:Distribution{Univariate,S<:ValueSupport}},Real) at /Users/bdeonovic/.julia/v0.3/Distributions/src/truncate.jl:39
is ambiguous with: 
    pdf(Distribution{Univariate,Discrete},Int64) at /Users/bdeonovic/.julia/v0.3/Distributions/src/univariates.jl:127.
To fix, define 
    pdf(Truncated{Discrete,D<:Distribution{Univariate,S<:ValueSupport}},Int64)
before the new definition.
Warning: New definition 
    pdf(Truncated{S<:ValueSupport,D<:Distribution{Univariate,S<:ValueSupport}},Real) at /Users/bdeonovic/.julia/v0.3/Distributions/src/truncate.jl:39
is ambiguous with: 
    pdf(Distribution{Univariate,Discrete},Integer) at /Users/bdeonovic/.julia/v0.3/Distributions/src/univariates.jl:128.
To fix, define 
    pdf(Truncated{Discrete,D<:Distribution{Univariate,S<:ValueSupport}},Integer)
before the new definition.
...
```